check-rules/checkstyle-rules.xml

```diff
@@ -0,0 +1,109 @@
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+    <property name="charset" value="UTF-8"/>
+
+    <property name="fileExtensions" value="java, properties, xml"/>
+
+    <!-- Whitespaces Checks. See: http://checkstyle.sourceforge.net/config_whitespace.html -->
+    <module name="FileTabCharacter">
+        <property name="eachLine" value="true"/>
+    </module>
+
+    <module name="TreeWalker">
+        <!-- Coding Checks. See: https://checkstyle.sourceforge.io/config_coding.html -->
+        <module name="SimplifyBooleanExpression"/>
+
+        <!-- Import Checks. See: http://checkstyle.sourceforge.net/config_imports.html -->
+        <module name="UnusedImports"/>
+
+        <!-- Whitespaces: See: https://checkstyle.sourceforge.io/config_whitespace.html -->
+        <module name="WhitespaceAround">
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+            <property name="allowEmptyTypes" value="true"/>
+            <property name="allowEmptyLoops" value="true"/>
+            <property name="allowEmptyLambdas" value="true"/>
+            <property name="allowEmptyCatches" value="true"/>
+        </module>
+
+        <!--Modifiers Checks. See: http://checkstyle.sourceforge.net/config_modifier.html-->
+        <module name="ModifierOrder"/>
+
+        <!--Annotation checks. See: http://checkstyle.sourceforge.net/config_annotation.html-->
+        <module name="MissingOverride"/>
+
+        <!-- Empty line checks. See: http://checkstyle.sourceforge.net/config_whitespace.html#EmptyLineSeparator -->
+        <module name="EmptyLineSeparator">
+            <!-- PACKAGE_DEF excluded according to Ignite code style. -->
+            <property name="tokens" value="IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF, STATIC_INIT,
+                INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF" />
+
+            <property name="allowMultipleEmptyLines" value="false" />
+            <property name="allowMultipleEmptyLinesInsideClassMembers" value="false" />
+        </module>
+
+        <!-- @Override annotations on the same line with a method declaration check. -->
+        <!--module name="org.apache.ignite.tools.checkstyle.OverrideAnnotationOnTheSameLineCheck"/-->
```

Comment:
Why this check is disabled?

Reply:
This check is disabled because there is no such class currently in Ignite 3.0 codebase (due to currently missing ignite-tools module). The check will be reenabled after ignite-tools are back in code.

Reply:
👍